src/highlight/HighlightCreator.tsx

```diff
@@ -1,11 +1,48 @@
-import * as React from 'react';
+import React from 'react';
 import classNames from 'classnames';
+import { getSelectionRange } from './highlightUtil';
 import './HighlightCreator.scss';
 
 type Props = {
     className?: string;
+    onSelectionChange?: (range: Range | null) => void;
+    pageEl: HTMLElement;
 };
 
-export default function HighlightCreator({ className }: Props): JSX.Element {
+export default function HighlightCreator({ className, onSelectionChange, pageEl }: Props): JSX.Element {
+    const handleMouseUp = (): void => {
+        if (!onSelectionChange) {
+            return;
+        }
+
+        setTimeout(() => {
+            onSelectionChange(getSelectionRange());
+        }, 300); // Pdf.js textLayer enhancement waits 300ms (yes, they hardcoded this magic number)
+    };
+
+    const handleSelectionChange = (): void => {
+        if (!onSelectionChange) {
+            return;
+        }
+
+        const selection = document.getSelection();
+
+        if (!selection || selection.isCollapsed) {
+            onSelectionChange(null);
+        }
+    };
+
+    React.useEffect(() => {
+        // Only document has selectionchange event. Document-level event listener
+        // allows the creator to respond even if the selection is on the other pages
+        document.addEventListener('selectionchange', handleSelectionChange);
```

Comment:
Is this still needed? I believe the decision is that only single page highlights will be supported

Reply:
This is still needed for canceling selections on the other pages. For example, page A had a selection and popup, when user selects text on page B, page A needs to know there is a new selection on the other pages and cancel the popup.